proxy-node-shared/src/main/java/uk/gov/ida/notification/validations/Base64XmlValidator.java

```diff
@@ -2,42 +2,29 @@
 
 import org.apache.commons.lang3.StringUtils;
 import org.glassfish.jersey.internal.util.Base64;
-import org.w3c.dom.Document;
-import org.xml.sax.SAXException;
 
 import javax.validation.ConstraintValidator;
 import javax.validation.ConstraintValidatorContext;
-import javax.xml.parsers.DocumentBuilder;
-import javax.xml.parsers.DocumentBuilderFactory;
-import javax.xml.parsers.ParserConfigurationException;
-import java.io.ByteArrayInputStream;
-import java.io.IOException;
-import java.io.InputStream;
 
 public class Base64XmlValidator implements ConstraintValidator<ValidBase64Xml, String> {
 
     @Override
     public void initialize(ValidBase64Xml constraint) { /* intentionally blank */ }
 
     @Override
-    public boolean isValid(String potentialXML, ConstraintValidatorContext context) {
+    public boolean isValid(String potentialBase64, ConstraintValidatorContext context) {
         // Detecting nulls, empties and whitespace is the responsibility of other validations.
         // Responding true here indicates that this validator does not have an opinion about empty values.
-        if (StringUtils.isBlank(potentialXML)) { return true; }
+        if (StringUtils.isBlank(potentialBase64)) { return true; }
 
-        String xml = Base64.decodeAsString(potentialXML);
-        if (StringUtils.isEmpty(xml)) { return false; }
-
-        InputStream stream = new ByteArrayInputStream(xml.getBytes());
-        DocumentBuilder parser = null;
         try {
-            parser = DocumentBuilderFactory.newInstance().newDocumentBuilder();
-            Document document = parser.parse(stream);
-            return document != null;
+            String decoded = Base64.decodeAsString(potentialBase64);
+            if (StringUtils.isBlank(decoded)) { return false; }
+            return decoded.trim().startsWith("<?xml version=");
```

Comment:
Is this enough to decide that the entire XML tree is valid?

Reply:
No but I cannot validate the tree. Feedback was not to operate on SAML or XML here. I had a validator that does that, but I removed it. This, at least, ensures that it's a base64 string...

Reply:
ah yes, the no parsing XML except in one place thing.